stdnum/at/stnr.py

```diff
@@ -0,0 +1,240 @@
+# -*- coding: utf-8 -*-
+# stnr.py - functions for handling Austrian tax number
+# coding: utf-8
+#
+# Copyright (C) 2017 Holvi Payment Services Oy
+# Copyright (C) 2017 Arthur de Jong
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301 USA
+""" steuernummer (Austrian tax number)
+A number issued by a tax office in Austria to tax businesses, the number is
+9 numerical digits long. The first two digits represent the tax office which
+issued the number. The latter 7 digits represent the tax id.
+https://dienstgeber.ooegkk.at/portal27/dgooegkkportal/content/contentWindow?viewmode=content&contentid=10007.681672
+The tax offices specified in a list on the tax authority website here:
+https://service.bmf.gv.at/Service/Anwend/Behoerden/show_mast.asp?Typ=SM&DisTyp=FA
+>>> validate('38/1234567')
+'381234567'
+>>> validate(number='38/1234567', office='Bruck Eisenstadt Oberwart')
+'381234567'
+>>> is_valid('38/1234567')
+True
+>>> is_valid(number='38/1234567', office='Bruck Eisenstadt Oberwart')
+True
+"""
+from stdnum.exceptions import *
+from stdnum.util import clean
+
+tax_offices_codes = {
```

Comment:
Do we use tax_offices_codes anywhere except get_all_codes_dict? May be we should store them in {tax_code: tax_office_info} format, so we can skip use get_all_codes_dict? Regions can be mentioned in comments

Reply:
I was thinking that we can reuse that dictionary in the front end in the future instead of collecting the information again. It is easier to deduct the information that you don't have than adding the information you are missing. For example, in our application we are not that specific to check for the state, maybe someone else will use that in the future.